lib/utils/src/graph/algorithms.cc

```diff
@@ -16,6 +16,13 @@ std::vector<Node> add_nodes(IGraph &g, int num_nodes) {
   return nodes;
 }
 
+std::vector<Node> add_nodes(DiGraph &g, int num_nodes) {
+  std::vector<Node> nodes;
+  std::generate_n(
```

Comment:
While I appreciate the functional syntax, I think a for loop is clearer here :slightly_smiling_face:

Reply:
OK.